tests/tcg/arc64/check_qmach.S

```diff
@@ -0,0 +1,553 @@
+;                  8257284
+  prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000000
+  set_reg64  r7, hi=0x7fff7fff, lo=0x7fff7fff
+  qmach    r6, r7, 63
+  flg_none_set
+  chk_reg64  r6, hi=0x00000000, lo=0x007dff04
+
+; Test case 14
+; reg <- reg, s12 (s12 is sign extended to 16 bits and then replicated 4 times)
+;   32767     32767    -32768    -32768
+; (0x7fff,   0x7fff,   0x8000,   0x8000) * r6
+; (    -1,       -1,       -1,       -1)   s12
+; (     0x0000_0000_0000_0000    )   r58 (acc)
+; --------------------------------
+; (   -32767 * 2 + 32768 * 2    )
+; -------------------------------
+;    0x0000_0000_0000_0002
+  prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000000
+  set_reg64  r6, hi=0x7fff7fff, lo=0x80008000
+  qmach    r6, r6, -1
+  flg_none_set
+  chk_reg64  r6, hi=0x00000000, lo=0x00000002
+
+; Test case 15
+; reg <- reg, s12 (smallest negative s12)
+;   32767     32767    -32768    -32768
+; (0x7fff,   0x7fff,   0x8000,   0x8000) * r9
+; ( -2048,    -2048,    -2048,    -2048)   s12
+; (     0x0000_0000_0000_0000    )   r58 (acc)
+; --------------------------------
+; (   -67106816 * 2 + 67108864 * 2    )
+; (   -134213632 + 134217728    )
+; -------------------------------
+;    0x0000_0000_0000_1000
+;                     4096
+  prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000000
+  set_reg64  r9, hi=0x7fff7fff, lo=0x80008000
+  qmach    r9, r9, -2048
+  flg_none_set
+  chk_reg64  r9, hi=0x00000000, lo=0x00001000
+
+; Test case 16
+; reg <- reg, s12 (biggest positive s12)
+;   32767     32767    -32768    -32768
+; (0x7fff,   0x7fff,   0x8000,   0x8000) * r7
+; (  2047,     2047,     2047,     2047)   s12
+; (     0x0000_0000_0000_0000    )   r58 (acc)
+; --------------------------------
+; (   67074049 * 2 + -67076096 * 2    )
+; (   -134213632 + 134217728    )
+; -------------------------------
+;    0xffff_ffff_ffff_f002
+;                    -4096
+  prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000000
+  set_reg64  r7, hi=0x7fff7fff, lo=0x80008000
+  qmach    r7, r7, 2047
+  flg_none_set
+  chk_reg64  r7, hi=0xffffffff, lo=0xfffff002
+
+
+; Test case 17
+; check .cc when it is not met, with .f
+; (  0,  0,  0,  0) * r7
+; (  0,  0,  0,  0)   s12
+; (     0x0000_0000_0000_0001    )   r58 (acc)
+; --------------------------------
+; set n flag
+; ----------------
+;  0 * 4 + 1 = 1 and n flag unset
+; ----------------
+; conditions not met, n flag should still be set
+;
+prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000001
+  set_reg64  r7, hi=0x00000000, lo=0x00000000
+  set_n_flag
+  qmach.p.f    r7, r7, 0
+  flg_n_set
+
+; Test case 18
+; check .cc when it is met, as well as .f
+; (  0,  0,  0,  0) * r7
+; (  0,  0,  0,  0)   s12
+; (     0x0000_0000_0000_0001    )   r58 (acc)
+; --------------------------------
+; set n flag
+; ----------------
+;  0 * 4 + 1 = 1 and n flag unset
+; ----------------
+; conditions are met, n flag should not be set
+;
```

Comment:
remove this (442) empty line. I see a pattern here. please take care of the rest of the file.